### PR TITLE
Prevent runner from running the jobs.

### PR DESCRIPTION
**Pull request by sarrangan**
This is needed in order to use bullmq in a similar way to bull.

### CODE REVIEW

**Review by csingh — CHANGES_REQUESTED** [review]
@sarrangan TESTS ARE FAILING WAT U DO

**Comment by sarrangan** [comment]
I HACKED IT

**Comment by sarrangan** [comment]
I UNHAcKED IT

File: src/interfaces/advanced-options.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 export interface AdvancedOptions {
   // A set of custom backoff strategies keyed by name.
   backoffStrategies?: {};
+  disableRun?: boolean;
```
Comment (csingh) [inline src/interfaces/advanced-options.ts]: Maybe `disableAutoRun` to be extra explicit? :) And also can you add a quick tidbit to the Changelog :D

File: src/classes/worker.ts [hunk 2]
```diff
@@ -77,6 +77,13 @@ export class Worker<T = any> extends QueueBase {
     }
     this.timerManager = new TimerManager();
 
+    // We need the run part of the code to be disabled
+    // in order to use bullMQ the way we expect it to
+    // in BullWhip. Using the setting below we can skip
+    // the run function when we need to
+    if (this.opts.settings && this.opts.settings.disableRun) {
```
Comment (csingh) [inline src/classes/worker.ts]: Can we only skip the run part based on `disableRun` instead of this return? :D

**Review by csingh — CHANGES_REQUESTED** [review]
couple commments

**Comment by sarrangan** [comment]
Couldn't put the commit hash for the feature in the Changelog, because we'll only get that once its like squashed and merged =(

File: src/classes/worker.ts [hunk 3]
```diff
@@ -77,10 +77,12 @@ export class Worker<T = any> extends QueueBase {
     }
     this.timerManager = new TimerManager();
 
-    /* tslint:disable: no-floating-promises */
-    this.run().catch(error => {
-      console.error(error);
-    });
+    if (!this.opts.settings || !this.opts.settings.disableAutoRun) {
```
Comment (csingh) [inline src/classes/worker.ts:80]: Does this still work? :D

Wouldn't you want to make sure `this.opts.settings` exists first before checking `disableAutoRun`?

**Review by csingh — COMMENTED** [review]
One moar question!

File: src/classes/worker.ts (hunk 3 above)
Comment (csingh) [reply src/classes/worker.ts:80]: Makes me wonder if its also possible to run a quick unit test for this case :) Like add some jobs to the queue, create a worker, and ensure that jobs do/don't get moved to completed based on the setting

**Comment by sarrangan** [comment]
Got the test in here!!!!

File: src/classes/worker.ts (hunk 3 above)
Comment (sarrangan) [reply src/classes/worker.ts:80]: Yup. `!this.opts.settings` basically returns true if `this.ops.settings` does not exist. So then the `or` condition early evaluates to true. If `this.opts.settings` does exist, then `!this.opts.settings` will evaluate to false and we'll check the next condition. The way I switched it though is using `A && B == !A || !B`. Added a test to be sure.

**Review by csingh — APPROVED** [review]
SHIP IT 🚢

**Pull request merged**